kiota_authentication_azure/_observability.py

```diff
@@ -0,0 +1,33 @@
+from kiota_http.middleware.parameters_name_decoding_handler import ParametersNameDecodingHandler
+from kiota_http.observability_options import ObservabilityOptions
+from opentelemetry import trace
+
+from ._version import VERSION
+
+tracer = trace.get_tracer(ObservabilityOptions.get_tracer_instrumentation_name(), VERSION)
```

Comment:
According to the document https://github.com/microsoftgraph/msgraph-sdk-design/blob/9974f7ed0baa99aeef92d19098bd239696c697db/Observability.md it is a span in the http library tracer. Is that a correct interpretation of the document?

Reply:
Not sure about that, but in other language implementations each library defines it's own observability options class

Dotnet
https://github.com/microsoft/kiota-authentication-azure-dotnet/blob/main/src/ObservabilityOptions.cs
https://github.com/microsoft/kiota-http-dotnet/blob/main/src/ObservabilityOptions.cs

Typescript
https://github.com/microsoft/kiota-typescript/blob/main/packages/authentication/azure/src/observabilityOptions.ts
https://github.com/microsoft/kiota-typescript/blob/main/packages/http/fetch/src/observabilityOptions.ts

Reply:
Here we're using the package name for the tracer name as you rightfully identified the fact that otherwise it'd look like this came from HTTP which would be misleading.
https://github.com/microsoft/kiota-authentication-azure-go/blob/2c624235eac546abe8ada7a4f57428b92f5d6f84/azure_identity_access_token_provider.go#L32

Reply:
I have updated it with this approach.